Cabal/src/Distribution/Simple/PreProcess.hs

```diff
@@ -292,10 +300,11 @@ preprocessFile
   -- ^ fail on missing file
   -> IO ()
 preprocessFile mbWorkDir searchLoc buildLoc forSDist baseFile verbosity builtinSuffixes handlers failOnMissing = do
+  debug verbosity $ "preprocessFile: " <> prettyShow baseFile
+  bsrcFiles <- findFileCwdWithExtension mbWorkDir builtinSuffixes (searchLoc ++ [buildAsSrcLoc]) baseFile
```

Comment:
First, we try and find a regular Haskell file.

We also add `buildAsSrcLoc]` to the end of the list, since the locations are tried in-order. Most likely, we're going to be in the first directory we're searching in, so this should eliminate a lot of file lookups.

Reply:
I wonder if this is a change in behaviour? Before, if you had a `.hs` and a `.y` file in the directory, would it prefer to regenerate the `.hs` from the `.y` file? 

It seems that it would then prefer the generated `.y` file rather than the `.hs` file in the directory, because it searched the build directory first?

Could we have some tests to check this behaviour?

Reply:
That's a good question - what *is* the expected behavior of having multiple files with different extensions in directory? Rather, what *should* happen if we have a directory project like this:

```
src/Foo.hs
src/Foo.y
```

The old code would *ignore* `src/Foo.hs` in favor of `src/Foo.y`. The new code ignores `src/Foo.y` in favor of `src/Foo.hs`.

This is definitely a change in behavior, but it would require a project to be poorly structured - somehow having `Foo.hs` and `Foo.y` together, with the expectation that `Foo.hs` is never used.